src/x86/nt.rs

```diff
@@ -0,0 +1,100 @@
+//! Non-temporal store operations.
+use core::marker::PhantomData;
+
+#[cfg(target_arch = "x86")]
+use core::arch::x86::{self as arch, __m256i, _mm_sfence};
+#[cfg(target_arch = "x86_64")]
+use core::arch::x86_64::{self as arch, __m256i, _mm_sfence};
+
+/// Load from a 32-bit aligned address with non-temporal hint, avoiding filling the cache.
+#[inline]
+#[target_feature(enable = "avx2")]
+pub fn _mm256_stream_load_si256(addr: &__m256i) -> __m256i {
+    unsafe { arch::_mm256_stream_load_si256(addr) }
+}
+
+/// Store into a 32-bit aligned address with non-temporal hint, avoiding clobbering the cache.
+#[inline]
+#[target_feature(enable = "avx2")]
+pub fn _mm256_stream_store_256i(addr: &mut NonTemporalStoreable<'_, __m256i>, v: __m256i) {
+    unsafe { arch::_mm256_stream_si256(addr.inner, v) }
+}
```

Comment:
Here and anywhere else the store is used, this is only target feature `avx`
https://doc.rust-lang.org/stable/core/arch/x86/fn._mm256_stream_si256.html

Load is `avx2`
https://doc.rust-lang.org/stable/core/arch/x86/fn._mm256_stream_load_si256.html